internal/gengapic/testdata/deprecated_client_init.want

```diff
@@ -131,15 +131,39 @@ func (c *gRPCClient) Close() error {
 
 // Methods, except Close, may be called concurrently. However, fields must not be modified concurrently with method calls.
 type restClient struct {
-	host string
+	endpoint string
+	httpClient http.Client
 }
 
 func (c *restClient) Zip(ctx context.Context, req *mypackagepb.Bar, opts ...gax.CallOption) (*mypackagepb.Foo, error) {
-	m := jsonpb.Marshaler{}
-	if jsonReq, err := m.MarshalToString(req); err != nil {
+	// The default (false) for the other options are fine.
+	marshaler := protojson.MarshalOptions{AllowPartial: true, EmitUnpopulated: true}
```

Comment:
Should we AllowPartial? This will just fail when it hits the backend then, right?

Reply:
It's already the backend's responsibility to check, is it not?

Reply:
This may be a `proto2` hold-over. `proto3` doesn't have `required`, and the `AllowPartial` docs say this `allows messages that have missing required fields to marshal without returning an error`. There'd be no use for this in `proto3` syntax since there is no `required`

Reply:
> It's already the backend's responsibility to check, is it not?

Backend would fail, but I think it is generally better to fail fast and save sending an RPC to figure this out.

> This may actually only matter for proto2,

If it is I don't think we need to set it then.